src/App.tsx

```diff
@@ -1,26 +1,218 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { FormEvent, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  creatTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+import { Header } from './conponents/Header/Header';
+import { TodoList } from './conponents/TodoList/TodoList';
+import { Footer } from './conponents/Footer/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<Status>(Status.All);
+  const [error, setError] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [newTitle, setNewTitle] = useState<string>('');
+
+  const titleField = useRef<HTMLInputElement>(null);
```

Comment:
These are the internals of the Header component - please move this state there
This would have several positive outcomes, the one is that as of now the whole app is re-rendered on each new character entered into the input